### PR TITLE
fix: metrics test is independent from execution of unrelated tests

### PR DESCRIPTION
**Pull request by nstng**
Signed-off-by: Nils Semmelrock <nils.semmelrock@tngtech.com>

## Summary

#14351 moved `test_gateway_metrics_attach_detach.py` in the execution order. This made an early assert fail because no `mme_new_association` event happened yet (see, e.g., https://github.com/magma/magma/runs/9277319986). Another attach/detach test needs to be executed firs, e.g., `test_attach_detach.py` if mme was restarted in the mean time.

Here: move the assert to the end of the test and check that the event counter actually increased during the test - dynamically so that possible events from other tests do not matter.

## Test Plan

Execute `test_gateway_metrics_attach_detach.py` after a freshly restarted mme.

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: lte/gateway/python/integ_tests/s1aptests/test_gateway_metrics_attach_detach.py [hunk 1]
```diff
@@ -39,11 +39,10 @@ def test_gateway_metrics_attach_detach(self):
         """ Basic gateway metrics with attach/detach for a single UE """
 
         label_values_success = {"result": "success"}
-        mme_new_association = self._getMetricValueGivenLabel(
+        mme_new_association_before = self._getMetricValueGivenLabel(
             "mme_new_association",
             label_values_success,
         )
-        assert mme_new_association > 0
```
Comment (crasu) [inline lte/gateway/python/integ_tests/s1aptests/test_gateway_metrics_attach_detach.py:46]: why did you remove this? Was this 0 when only this test was exxecuted?

File: lte/gateway/python/integ_tests/s1aptests/test_gateway_metrics_attach_detach.py [hunk 2]
```diff
@@ -138,6 +137,12 @@ def test_gateway_metrics_attach_detach(self):
             )
             assert (val == v_spgw_delete_session + 1)
 
+        mme_new_association = self._getMetricValueGivenLabel(
+            "mme_new_association",
+            label_values_success,
+        )
+        assert (mme_new_association == mme_new_association_before + 1)
```
Comment (crasu) [inline lte/gateway/python/integ_tests/s1aptests/test_gateway_metrics_attach_detach.py:144]: What does this counter measure? I would have assumed that this value is 2 after the test.

**Review by crasu — COMMENTED** [review]
A description in the pr instead of TBD be nice

**Review by crasu — APPROVED** [review]

File: lte/gateway/python/integ_tests/s1aptests/test_gateway_metrics_attach_detach.py (hunk 1 above)
Comment (nstng) [reply lte/gateway/python/integ_tests/s1aptests/test_gateway_metrics_attach_detach.py:46]: see description

File: lte/gateway/python/integ_tests/s1aptests/test_gateway_metrics_attach_detach.py (hunk 2 above)
Comment (nstng) [reply lte/gateway/python/integ_tests/s1aptests/test_gateway_metrics_attach_detach.py:144]: I am not completely sure ... I started my local test with +2, but it seems to be +1. The doc says "New SCTP association". I.e., maybe something like a session based counter.

Comment (crasu) [reply lte/gateway/python/integ_tests/s1aptests/test_gateway_metrics_attach_detach.py:144]: If these are sctp association 1 makes sense.

Comment (rsarwad) [reply lte/gateway/python/integ_tests/s1aptests/test_gateway_metrics_attach_detach.py:144]: Yes, on new sctp association, for label "mme_new_association" count gets incremented.

File: lte/gateway/python/integ_tests/s1aptests/test_gateway_metrics_attach_detach.py [hunk 3]
```diff
@@ -39,11 +39,10 @@ def test_gateway_metrics_attach_detach(self):
         """ Basic gateway metrics with attach/detach for a single UE """
 
         label_values_success = {"result": "success"}
-        mme_new_association = self._getMetricValueGivenLabel(
+        mme_new_association_before = self._getMetricValueGivenLabel(
```
Comment (rsarwad) [inline lte/gateway/python/integ_tests/s1aptests/test_gateway_metrics_attach_detach.py:42]: For every test case, new sctp association is created as part Setup() and every test case has Teardown() when test cases exits. In TearDown(), s1ap tester sends sctp shutdown which eventually clears the association. As part of mme restart, is connection closed?

Comment (nstng) [reply lte/gateway/python/integ_tests/s1aptests/test_gateway_metrics_attach_detach.py:42]: @rsarwad aww sorry, I already mergede the PR - do you see an issue with this? Then I will create a new PR.
My observation was that the metric counter remained after `test_attach_detach.py` was executed and `test_gateway_metrics_attach_detach` actually expected this by an `assert counter > 0`. I.e., I would expect that `test_attach_detach.py` does not clean up the counter correctly.

Comment (rsarwad) [reply lte/gateway/python/integ_tests/s1aptests/test_gateway_metrics_attach_detach.py:42]: Looks like counter for label "mme_new_association" is not getting decremented on sctp shutdown at mme. Changes in PR make sure that only one association is created before test case terminates
